File: web/src/models/Bill.ts

export type User = {
    id: number
    email: string
    fullName: string
}

export type Order = {
    id: number
    item_id: number
    qty: number
}

export type Participant = {
    id: number
    user: User
    orders: Order[]
}

export type Item = {
    id: number
    name: string
    price: number
    initialQty: number
}

export type Bill = {
    id: number
    code: string
    title: string
    description: string
    participants: Participant[]
    items: Item[]
}

File: entities/bill.go
package entities

type Bill struct {
	ID           int64         `json:"id"`
	Code         string        `json:"code"`
	Title        string        `json:"title"`
	Description  string        `json:"description"`
	Host         *User         `json:"host,omitempty"`
	Participants []Participant `json:"participants,omitempty"`
	Items        []Item        `json:"items,omitempty"`
}
